lib/app/cli.ts

```diff
@@ -0,0 +1,249 @@
+    }
+
+    // Check if we have been cloned and not downloaded
+    if (fs.existsSync('.git')) {
+        return child_process.execSync('git rev-parse HEAD').toString().trim();
+    }
+
+    // unknown case
+    return '';
+}
+
+/**
+ * Extract release build number from file
+ */
+export function getReleaseBuildNumber(distPath: string, isDist: boolean): string {
+    // Use the canned build only if provided
+    const releaseBuildPath = path.join(distPath, 'release_build');
+    if (isDist && fs.existsSync(releaseBuildPath)) {
+        return fs.readFileSync(releaseBuildPath).toString().trim();
+    }
+    return '';
+}
+
+/**
+ * Detect if running under Windows Subsystem for Linux
+ */
+export function detectWsl(): boolean {
+    if (process.platform === 'linux') {
+        try {
+            return child_process.execSync('uname -a').toString().toLowerCase().includes('microsoft');
+        } catch (e) {
+            logger.warn('Unable to detect WSL environment', e);
+        }
+    }
+    return false;
+}
+
+/**
+ * Set up temporary directory, especially for WSL environments
+ */
+export function setupTempDir(options: CompilerExplorerOptions, isWsl: boolean): void {
+    // If a tempDir is supplied, use it
+    if (options.tmpDir) {
+        if (isWsl) {
+            process.env.TEMP = options.tmpDir; // for Windows
+        } else {
+            process.env.TMP = options.tmpDir; // for Linux
+        }
+    }
+    // If running under WSL without explicit tmpDir, try to use Windows %TEMP%
+    else if (isWsl) {
+        try {
+            const windowsTemp = child_process.execSync('cmd.exe /c echo %TEMP%').toString().replaceAll('\\', '/');
+            const driveLetter = windowsTemp.substring(0, 1).toLowerCase();
+            const directoryPath = windowsTemp.substring(2).trim();
+            process.env.TEMP = path.join('/mnt', driveLetter, directoryPath);
+        } catch (e) {
+            logger.warn('Unable to invoke cmd.exe to get windows %TEMP% path.');
+        }
+    }
+}
+
+/**
+ * Convert command-line options to AppArguments for the application
+ */
+export function convertOptionsToAppArguments(
+    options: CompilerExplorerOptions,
+    gitReleaseName: string,
+    releaseBuildNumber: string,
+): AppArguments {
+    return {
+        rootDir: options.rootDir,
+        env: options.env,
+        hostname: options.host,
+        port: options.port,
+        gitReleaseName: gitReleaseName,
+        releaseBuildNumber: releaseBuildNumber,
+        wantedLanguages: options.language,
+        doCache: options.cache,
+        fetchCompilersFromRemote: options.remoteFetch,
+        ensureNoCompilerClash: options.ensureNoIdClash,
+        suppressConsoleLog: options.suppressConsoleLog,
+    };
+}
+
+/**
+ * Initialize configuration from command-line arguments
+ * @param argv The command-line arguments to parse
+ * @returns Application arguments and raw options
+ */
+export function initializeOptionsFromCommandLine(argv: string[]): {
```

Comment:
Wouldn't it make more sense if the logger setup instead took the options/app args?? etc?

Reply:
There's no consolidated log setup to pass to: I'll create one

Reply:
And also untangle the start up and hopefully avoid the `opts` ever leaking! Thanks for suggesting..